FSW/src/util/faultManager.cpp

```diff
@@ -0,0 +1,303 @@
+/* faultManager.cpp handles NS2 payload fault mitigation
+ * Usage:
+ *  module functionality
+ *  function definitions
+ *  put function declarations in memUtil.hpp
+ * 
+ * Modules encompassed:
+ *  Fault Detection
+ *  Fault Logging
+ *  Fault Mitigation
+ *
+ * Additional files needed for compilation:
+ *  config.hpp
+ *  encodedFile.hpp
+ *  
+ */
+
+/* - - - - - - Includes - - - - - - */
+// All libraries are put in dataCollection.hpp
+// NS2 headers
+#include "../headers/faultManager.hpp"
+#include "../headers/encodedFile.hpp"
+#include "../headers/commandHandling.hpp"
+#include "../headers/housekeeping.hpp"
+
+/* Module Variable Definitions */
+// fault log
+static FaultReport faultLog[faultCode::COUNT];
+// decoded size of data on EEPROM, in terms of bytes
+const size_t EEPROM_DECODED_SIZE = faultCode::COUNT * FaultReport::MEMSIZE + PayloadData::MEMSIZE; 
+// flag to indicate whether any faults were logged in the current main loop iteration
+static bool detectedFault = false; 
+
+/* - - - - - - Module Driver Functions - - - - - - */
+
+/* - - - - - - logFault - - - - - - *
+ * Usage:
+ *  records a new occurence of a fault
+ *  
+ * Inputs:
+ *  code - fault code to log
+ * 
+ * Outputs:
+ *  None
+ */
+void logFault(uint8_t code) {
+    detectedFault = true;
+
+    // check if fault is valid
+    if (code >= faultCode::ERR_CODE) {
+        Serial.println("Warning, tried to log a fault with invalid fault code (Fault Manager)");
```

Comment:
Perhaps print the invalid fault code with this, to identify what the invalid fault logged was.

Reply:
Done